### DISCUSSION TITLE
Lightsleep not working on Raspberry Pi Pico W

### DISCUSSION THREAD

**Discussion by jotonedev** [micropython/micropython#11159] [RP2040 / Pico]
When I try to put the pico w in lightsleep for X milliseconds, it stops responding.

The code I'm trying to execute:
```python
import machine
machine.lightsleep(50)
```

**Comment by robert-hh** [comment]
Try to update the firmware to the most recent nightly build, if you did not do that already. That behavior was fixed.

**Reply by jotonedev** [reply]
Tried with `v1.19.1-994-ga4672149b` but still not working correctly.

**Comment by robert-hh** [comment]
Is it possible that scenario from this PR applies: https://github.com/micropython/micropython/pull/10803

**Reply by jotonedev** [reply]
I've compiled that fork and now only some times it works and only when I choose low time periods like 1-5 milliseconds.

**Reply by jotonedev** [reply]
To test it I used this code

```python
from machine import lightsleep

print("test 1")
lightsleep(1)

print("test 2")
lightsleep(2)

print("test 3")
lightsleep(3)

print("test 4")
lightsleep(4)

print("test 5")
lightsleep(5)
```
It seems that `lightsleep(1)` always works.
